uw_sws/enrollment.py

```diff
@@ -49,11 +50,15 @@ def get_enrollment_by_regid_and_term(regid, term):
                                 term.year,
                                 term.quarter,
                                 regid)
-    return _json_to_enrollment(get_resource(url))
+    return _json_to_enrollment(get_resource(url), url)
```

Comment:
Ethan,
+ if "StatusCode" in json_data and json_data["StatusCode"] == "404":
+       raise DataFailureException(url, 404, json_data["StatusDescription"])
this piece is not right.  you may want to roll back the change in enrollment.py .